lib/nodeo/Antenna.js

```diff
@@ -14,140 +25,212 @@ var  y1=0;
+      x1 = x1+h;
+      asdf = array_factorcir(x2, x1, phi_desired, distance, dim);
+      y[i] = Math.abs(asdf*asdf*Math.sin(x1-pi/2));
+  }
+
+  var s=0;
+  for (i=0; i<N1+1; i++) {
+    if (i==0 || i==N1) {
+	    s=s+y[i];
+    } else {
+	    s=s+2*y[i];
+    }
+  }
 
-var pi=3.141592654;
-var dim=length(x1);
-var y=0;
-var num_nullMatrix=nullMatrix.length;
-var num1=300;
+ return (h/2)*s;
+}
 
-var phi = []
+//////////////////////////////////////////////////////////////
+/* This function calculates the fitness value of array 'x1' which is returned in 'y'
+The returned result is a  ans  object, as defined in TODO */
+function antennafunccircular(x1, nullMatrix, phi_desired, distance) {
+/* x1          -> Vector/Array
+   nullMatrix  -> Vector/Array
+   phi_desired -> Number/Scalar
+   distance    -> Number/Scalar
+*/
+  //En Matlab,  directivity  era global
+  var directivity;
+  var pi=3.141592654;
+  var dim=x1.length;
+  var y=0;
+  var num1=300;
+  var sidelobes = [];
+  var sllphi = [];
 
-//TODO Crear un vector linea de 0 a 360 con num1 elementos
-phi=linspace(0,360,num1);
-var phizero=0;
-yax = []
-yax[0]=array_factorcir(x1,(pi/180)*phi[0],phi_desired,distance,dim);
-var maxi=yax;
-var phi_ref=1;
-//for i=2:num1//This loop finds out the maximum gain 
-for(i=1;i<num1;i++){
-    yax[i]=array_factorcir(x1,(pi/180)*phi[i],phi_desired,distance,dim);
-    if (maxi<yax[i]){
-        maxi=yax[i];
-        phizero=phi[i];
-        phi_ref=i;
-    }
+  var upper_bound=180;
+  var lower_bound=180;
+
+  var maxtem=0;
+  var count=-1;
+
+  var phi = [0];
+  for (var i=1; i<num1; i++) {
+      phi.push(i*360/(num1-1));
+  }
+
+  var phizero=0;
+  var phi_ref=0; // or 1?
+
+  var yax = [];
+  var maxi;
+
+  yax[0] = array_factorcir(x1, (pi/180)*phi[0], phi_desired, distance, dim);
+  maxi=yax[0];
+
+  //This loop finds out the maximum gain 
+  for(var i=1; i<num1; i++) {
+      yax[i] = (array_factorcir(x1, (pi/180)*phi[i], phi_desired, distance, dim));
+      if (maxi<yax[i]) {
+          maxi=yax[i];
+          phizero=phi[i];
+          phi_ref=i;
+      }
+  }
+
+  if (yax[0]>yax[num1-1] && yax[0]>yax[1]) {
+      count = count+1;
+      sidelobes[count] = yax[0];
+      sllphi[count]=phi[0];
+  }
+
+  if (yax[num1-1]>yax[0] && yax[num1-1]>yax[num1-2]) {
+      count=count+1;
+      sidelobes[count]=yax[num1-1];
+      sllphi[count]=phi[num1-1];
+  }
+
+  for (var i=1; i<num1-1; i++) {
+      if (yax[i]>yax[i+1] && yax[i]>yax[i-1]) {
+          count=count+1;
+          sidelobes[count]=yax[i];
+          sllphi[count]=phi[i];
+      }
+  }
+
+  sidelobes.sort();
+  sidelobes.reverse();
+
+  y=sidelobes[1]/maxi;
+  var sllreturn=20*Math.log(y)/Math.log(10);
+
+  for (var i=0; i<num1/2; i++) {
+      if ((phi_ref+i)>num1-2) {
+          upper_bound=180;
+          break;
+      }
+      tem=yax[phi_ref+i];
+      if (yax[phi_ref+i]<yax[phi_ref+i-1] && yax[phi_ref+i]<yax[phi_ref+i+1]) {
+          upper_bound=phi[phi_ref+i]-phi[phi_ref];

```

Comment:
Identifier 'upper_bound' is not in camel case.<br>Identifier 'phi_ref' is not in camel case.